### PR TITLE
MAINT: Refactor digitation for KIT, artemis123, bti

### PR DESCRIPTION
**Pull request by massich**
This PR:
- takes over #5986, #6362.
- Refactors part of the digitization to make them look similar (for further refactor).

### CODE REVIEW

**Comment by agramfort** [comment]
looking good so far 👍

**Comment by larsoner** [comment]
Let me know when CIs are green and I should look

**Comment by massich** [comment]
@larsoner I'm AFK but Travis is green

**Comment by massich** [comment]
But coverage went off. I'll check

**Comment by massich** [comment]
Oh yes, I did not update the doc so it searches mne.io.DigPoint rather than mne.digitization.DigPoint

**Comment by massich** [comment]
Can I get feedback for this PR?

cc: @mne-tools/contributors

File: mne/digitization/_utils.py [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+                'Points have coord_frame entries that are incompatible with '
+                'coord_frame=%i: %s.' % (coord_frame, str(tuple(bad_frames))))
+
+    with start_file(fname) as fid:
+        write_dig_points(fid, pts, block=True, coord_frame=coord_frame)
+        end_file(fid)
+
+
+def _read_dig_points(fname, comments='%', unit='auto'):
+    """Read digitizer data from a text file.
```
Comment (agramfort) [inline mne/digitization/_utils.py]: apparently you can read more than from a .txt file

File: mne/digitization/_utils.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+    hpi : array-like | numpy.ndarray, shape (n_points, 3) | None
+        Points designated as head position indicator points.
+    extra_points : array-like | numpy.ndarray, shape (n_points, 3)
+        Points designed as the headshape points.
+    dig_ch_pos : dict
+        Dict of EEG channel positions.
+
+    Returns
+    -------
+    dig : Digitization
```
Comment (agramfort) [inline mne/digitization/_utils.py]: list ?

Comment (agramfort) [reply mne/digitization/_utils.py]: maybe you can already put a doc template for a parameter of type dig?

File: mne/digitization/_utils.py [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+# From mne.io.kit
+def _set_dig_kit(mrk, elp, hsp):
+    """Add landmark points and head shape data to the KIT instance.
+
+    Digitizer data (elp and hsp) are represented in [mm] in the Polhemus
+    ALS coordinate system. This is converted to [m].
+
+    Parameters
+    ----------
+    mrk : None | str | array_like, shape = (5, 3)
```
Comment (agramfort) [inline mne/digitization/_utils.py]: can you remove the = after shape in this docstring?

File: mne/digitization/_utils.py [hunk 4]
```diff
@@ -0,0 +1,395 @@
...
+        marker coils with respect to the digitized head shape, or path to a
+        file containing these points.
+    hsp : None | str | array, shape = (n_points, 3)
+        Digitizer head shape points, or path to head shape file. If more
+        than 10`000 points are in the head shape, they are automatically
+        decimated.
+
+    Returns
+    -------
+    dig_points : list
```
Comment (agramfort) [inline mne/digitization/_utils.py]: use a standard docstring

File: mne/digitization/_utils.py [hunk 5]
```diff
@@ -0,0 +1,395 @@
...
+        List of digitizer points for info['dig'].
+    dev_head_t : dict
+        A dictionary describe the device-head transformation.
```
Comment (agramfort) [inline mne/digitization/_utils.py]: do we have a template docstring for this?

File: mne/digitization/_utils.py [hunk 6]
```diff
@@ -0,0 +1,395 @@
...
+                             "%s." % (elp, elp_points.shape))
+        elp = elp_points
+    elif len(elp) != 8:
+        raise ValueError("ELP should contain 8 points; got shape "
+                         "%s." % (elp.shape,))
+    if isinstance(mrk, str):
+        mrk = read_mrk(mrk)
+
+    # hsp = apply_trans(als_ras_trans, hsp)
+    # elp = apply_trans(als_ras_trans, elp)
```
Comment (agramfort) [inline mne/digitization/_utils.py]: cleanup?

File: mne/digitization/_utils.py [hunk 7]
```diff
@@ -0,0 +1,395 @@
...
+    # move into MNE head coords
+    dig_points, _ = _call_make_dig_points(nas, lpa, rpa, hpi, extra)
+    return dig_points
+
+
+##############################################################################
+# From bti
+def _make_bti_dig_points(nasion, lpa, rpa, hpi, extra,
+                         convert=False, use_hpi=False,
+                         bti_dev_t=False, dev_ctf_t=False):
```
Comment (agramfort) [inline mne/digitization/_utils.py]: docstring?

File: mne/digitization/base.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+    a readable string representation.
+
+    Parameters
+    ----------
+    kind : int
+        Digitization kind, e.g. ``FIFFV_POINT_EXTRA``.
+    ident : int
+        Identifier.
+    r : ndarray, shape (3,)
+        Position.
```
Comment (agramfort) [inline mne/digitization/base.py]: position in meters?

File: mne/digitization/_utils.py (hunk 6 above)
Comment (agramfort) [reply mne/digitization/_utils.py]: it was there before? is this expected?

File: mne/digitization/_utils.py (hunk 4 above)
Comment (massich) [reply mne/digitization/_utils.py]: what is not standard? in this return?

File: mne/digitization/_utils.py (hunk 7 above)
Comment (massich) [reply mne/digitization/_utils.py]: I don't think is necessary this function should die in the next PR.

**Comment by massich** [comment]
I did not touch the docstrings much. (Just a find and replace of `dig: list of dicts` to `Digitization` and did not revert). I'll push a pass.

File: mne/digitization/_utils.py (hunk 5 above)
Comment (massich) [reply mne/digitization/_utils.py]: No I'll make a PR.

**Comment by agramfort** [comment]
ok to merge then when you're happy and CIs are green. I agree to do one thing at a time

**Comment by larsoner** [comment]
I probably can't really look until Friday so don't wait for me.

File: mne/digitization/_utils.py [hunk 9]
```diff
@@ -0,0 +1,393 @@
...
+from ..io.write import end_file
+from ..io.write import write_dig_points
+
+from ..transforms import apply_trans
+from ..transforms import als_ras_trans
+from ..transforms import get_ras_to_neuromag_trans
+from ..transforms import Transform
+from ..transforms import combine_transforms
+from ..transforms import invert_transform
+from ..transforms import _to_const
```
Comment (GuillaumeFavelier) [inline mne/digitization/_utils.py]: You prefer one `import` at a time?

File: mne/digitization/_utils.py [hunk 10]
```diff
@@ -0,0 +1,393 @@
...
+    hsp = apply_trans(nmtrans, hsp)
+
+    # device head transform
+    trans = fit_matched_points(tgt_pts=elp[3:], src_pts=mrk, out='trans')
+
+    nasion, lpa, rpa = elp[:3]
+    elp = elp[3:]
+
+    dig_points = _make_dig_points(nasion, lpa, rpa, elp, hsp)
+    dev_head_t = Transform('meg', 'head', trans)
```
Comment (GuillaumeFavelier) [inline mne/digitization/_utils.py]: To be consistent with previous call:
`dev_head_t = Transform(fro='meg', to='head', trans=trans)`

File: mne/digitization/_utils.py [hunk 11]
```diff
@@ -0,0 +1,393 @@
...
+    _hpi = hpi if use_hpi else None
+    info_dig, ctf_head_t = _call_make_dig_points(nasion, lpa, rpa, _hpi, extra,
+                                                 convert)
+
+    if convert:
+        t = combine_transforms(invert_transform(bti_dev_t), dev_ctf_t,
+                               'meg', 'ctf_head')
+        dev_head_t = combine_transforms(t, ctf_head_t, 'meg', 'head')
+    else:
+        dev_head_t = Transform('meg', 'head', trans=None)
```
Comment (GuillaumeFavelier) [inline mne/digitization/_utils.py]: `dev_head_t = Transform(fro='meg', to='head', trans=None)`

File: mne/digitization/_utils.py [hunk 12]
```diff
@@ -0,0 +1,393 @@
...
+def _write_dig_points(fname, dig_points):
+    """Write points to text file.
+
+    Parameters
+    ----------
+    fname : str
+        Path to the file to write. The kind of file to write is determined
+        based on the extension: '.txt' for tab separated text file.
+    dig_points : numpy.ndarray, shape (n_points, 3)
+        Points.
```
Comment (GuillaumeFavelier) [inline mne/digitization/_utils.py]: I'm not sure of the exact name but you seem to call it `DigPoints` or `Digitalization points` after.

File: mne/digitization/_utils.py [hunk 13]
```diff
@@ -0,0 +1,393 @@
...
+    """
+    _, ext = op.splitext(fname)
+    dig_points = np.asarray(dig_points)
+    if (dig_points.ndim != 2) or (dig_points.shape[1] != 3):
+        err = ("Points must be of shape (n_points, 3), "
```
Comment (GuillaumeFavelier) [inline mne/digitization/_utils.py]: Maybe `DigPoints` then?

File: mne/digitization/_utils.py (hunk 9 above)
Comment (massich) [reply mne/digitization/_utils.py]: we don't have a strict policy that I'm aware of, and its easier like this when I'm moving code around. I'll do a clean up on the last PR.

File: mne/digitization/_utils.py (hunk 10 above)
Comment (massich) [reply mne/digitization/_utils.py]: agreed. I'll at them on the next PR.

File: mne/digitization/_utils.py (hunk 5 above)
Comment (massich) [reply mne/digitization/_utils.py]: ￼I'll push them in #6385

**Pull request merged**